BetaUtils.Core/extensions/IEnumerableExtensions.cs

```diff
@@ -0,0 +1,106 @@
+
+        /// <summary>
+        /// Iterates through IEnumerable
+        /// </summary>
+        /// <typeparam name="items">IEnumerable</typeparam>
+        /// <typeparam name="action">Action</typeparam>
+        /// <returns>IEnumerable</returns>
+        public static IEnumerable<T> ForEach<T>(this IEnumerable<T> items, Action<T> action)
+        {
+            if (action is null)
+            {
+                throw new ArgumentNullException(nameof(action));
+            }
+
+            foreach (var item in items)
+            {
+                action(item);
+            }
+
+            return items;
+        }
+
+        /// <summary>
+        /// Converts an IEnumerable to a HashSet
+        /// </summary>
+        /// <typeparam name="items">IEnumerable</typeparam>
+        /// <returns>HashSet</returns>
+        public static HashSet<T> ToHashSet<T>(this IEnumerable<T> items)
+        {
+            HashSet<T> hashSet = new HashSet<T>();
+
+            foreach (var item in items)
+            {
+                hashSet.Add(item);
+            }
+
+            return hashSet;
+        }
+
+        /// <summary>
+        /// Converts an IEnumerable to an Observable Collection
+        /// </summary>
+        /// <param name="items">IEnumerable</param>
+        /// <returns>ObservableCollection</returns>
+        public static ObservableCollection<T> ToObservableCollection<T>(this IEnumerable<T> items)
+        {
+            var observableCollection = new ObservableCollection<T>();
+
+            foreach (var item in items)
+            {
+                observableCollection.Add(item);
+            }
+
+            return observableCollection;
+        }
+
+        /// <summary>
+        /// Takes a collection and splits it into a batch of collections with provided number of values
+        /// </summary>
+        /// <param name="items">IEnumerable</param>
+        /// <param name="numberOfItemsPerBatch">Integer</param>
+        /// <returns>New IEnumerable collections split into collections with provided number of items</returns>
+        public static IEnumerable<IEnumerable<T>> Batch<T>(this IEnumerable<T> items, int numberOfItemsPerBatch)
+        {
+            return items.Select((item, totalNumber) => new { item, totalNumber })
+                .GroupBy(x => x.totalNumber / numberOfItemsPerBatch)
+                .Select(x => x.Select(y => y.item));
+        }
+
+        /// <summary>
+        /// Takes IEnumerable as a source and returns it with all the null values removed
+        /// </summary>
+        /// <param items="source">IEnumerable</param>
+        /// <returns>IEnumerable without null values</returns>
+        /// <exception cref="ArgumentNullException"></exception>
+        public static IEnumerable<T> RemoveNulls<T>(this IEnumerable<T> items)
+        {
+            foreach (var item in items)
+            {
+                if (item != null)
+                {
+                    yield return item;
```

Comment:
There is another approach to this, since it is expecting IEnumerable as return value it could have been done with Linq like this: 

```c#
            return items.Where(item => item != null);
``` 
But this does not have any benefits over existing code (except its one line code) I will leave it to you to decide what approach you will take! 😁